src/4626/CompoundERC4626.sol

```diff
@@ -0,0 +1,201 @@
+/// @notice ERC4626 wrapper for Moonwell Finance
+contract CompoundERC4626 is ERC4626 {
+    /// -----------------------------------------------------------------------
+    /// Libraries usage
+    /// -----------------------------------------------------------------------
+
+    using LibCompound for MErc20;
+    using SafeTransferLib for ERC20;
+
+    /// -----------------------------------------------------------------------
+    /// Events
+    /// -----------------------------------------------------------------------
+
+    event ClaimRewards(uint256 amount, address token);
+
+    /// -----------------------------------------------------------------------
+    /// Errors
+    /// -----------------------------------------------------------------------
+
+    /// @notice Thrown when a call to Compound returned an error.
+    /// @param errorCode The error code returned by Compound
+    error CompoundERC4626__CompoundError(uint256 errorCode);
+
+    /// -----------------------------------------------------------------------
+    /// Constants
+    /// -----------------------------------------------------------------------
+
+    uint256 internal constant NO_ERROR = 0;
+
+    /// -----------------------------------------------------------------------
+    /// Immutable params
+    /// -----------------------------------------------------------------------
+
+    /// @notice The WELL token contract
+    ERC20 public immutable well;
+
+    /// @notice The Moonwell mToken contract
+    MErc20 public immutable mToken;
+
+    /// @notice The address that will receive the liquidity mining rewards (if any)
+    address public immutable rewardRecipient;
+
+    /// @notice The Compound comptroller contract
+    IComptroller public immutable comptroller;
+
+    /// -----------------------------------------------------------------------
+    /// Constructor
+    /// -----------------------------------------------------------------------
+
+    constructor(
+        ERC20 asset_,
+        ERC20 well_,
+        MErc20 mToken_,
+        address rewardRecipient_,
+        IComptroller comptroller_
+    ) ERC4626(asset_, _vaultName(asset_), _vaultSymbol(asset_)) {
+        well = well_;
+        mToken = mToken_;
+        comptroller = comptroller_;
+        rewardRecipient = rewardRecipient_;
+    }
+
+    /// -----------------------------------------------------------------------
+    /// Compound liquidity mining
+    /// -----------------------------------------------------------------------
+
+    /// @notice Claims liquidity mining rewards from Compound and sends it to rewardRecipient
+    function claimRewards() public {
+        address[] memory holders = new address[](1);
+        holders[0] = address(this);
+
+        MToken[] memory mTokens = new MToken[](1);
+        mTokens[0] = MToken(address(mToken));
+
+        comptroller.claimReward(holders, mTokens, false, true);
+
+        uint256 amount = well.balanceOf(address(this));
```

Comment:
I think we need to call `comptroller.rewardDistributor.getAllMarketConfigs()` here to get all the reward tokens and then do a loop through it just like we're doing in the `sweepRewards` method. Wdyt?

Reply:
It's not the same mToken that we have used 3 lines above?

Reply:
good catch